CMakeLists.txt

```diff
@@ -0,0 +1,115 @@
+
+project(
+OpenAssetIO-MediaCreation
+VERSION 1.0.0
+)
+
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+# Additional include directories for CMake utils.
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/cmake)
+
+#-----------------------------------------------------------------------
+# Options
+option(OPENASSETIO_MEDIACREATION_ENABLE_TEST "Run test on mediacreation traits" OFF)
+message(STATUS "Test enabled = ${OPENASSETIO_MEDIACREATION_ENABLE_TEST}")
+
+# ABI wrangling only needed for test
+if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND
+    CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 5.0)
+    option(OPENASSETIO_MEDIACREATION_GLIBCXX_USE_CXX11_ABI "For gcc, use the new C++11 library ABI"
+           OFF)
+    message(STATUS "New C++11 ABI for GCC = ${OPENASSETIO_MEDIACREATION_GLIBCXX_USE_CXX11_ABI}")
+endif ()
+
+#-----------------------------------------------------------------------
+# Default install directory
+
+# Default install to a `dist` directory under the build directory, ready
+# for use in tests and for packaging. But don't override if user has
+# explicitly set CMAKE_INSTALL_PREFIX.
+if (CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT AND PROJECT_IS_TOP_LEVEL)
+    set(CMAKE_INSTALL_PREFIX "${PROJECT_BINARY_DIR}/dist" CACHE PATH "Installation location" FORCE)
+endif ()
+
+#-----------------------------------------------------------------------
+# OpenAssetIO-MediaCreation generates its traits at configure time, via
+# calling `openassetio-traitgen`.
+# Some might be surprised by this, and expect code generation to be
+# performed in a build step. However, with MediaCreation being a
+# header only library, there is no "build-step" as such, (unless added
+# coincidently, if you are building the test executable, for example.)
+# Furthermore, generating sources at configure time serves to more
+# directly mimic a non-generated project, and lowers risk that changes
+# to the library strucutre later on will be caused issue by the
+# generative nature of the project.
+message("Generating Traits with OpenAssetIO-Traitgen")
+# Mark the traits source as a configure dependency by copying it to the
+# binary dir, and use that file to run traitgen from.
+# This means if the source traits file is changed, configure will rerun.
+configure_file(${CMAKE_CURRENT_LIST_DIR}/traits.yml ${CMAKE_CURRENT_BINARY_DIR}/traits.yml)
+execute_process(COMMAND openassetio-traitgen ${CMAKE_CURRENT_BINARY_DIR}/traits.yml
+                -o ${CMAKE_CURRENT_LIST_DIR} -g cpp)
+
+add_library(openassetio-mediacreation INTERFACE)
+# add alias so the project can be used with add_subdirectory
+add_library(OpenAssetIO-MediaCreation::openassetio-mediacreation ALIAS openassetio-mediacreation)
+
+# traitgen generates to _public_header_source_root location by default.
+set(_public_header_source_root "${CMAKE_CURRENT_SOURCE_DIR}/openassetio_mediacreation/include")
+set(_config_install_dir "${CMAKE_INSTALL_LIBDIR}cmake/${PROJECT_NAME}")
```

Comment:
[fix] This installs to `dist/cmake/OpenAssetIO-MediaCreation`, which is clearly not the intent.  We're missing the `include(GNUInstallDirs)`, and we're missing a `/`.

Reply:
Wasn't clear to me...
https://github.com/OpenAssetIO/OpenAssetIO-MediaCreation/pull/25/commits/c3892a62bc5f12495330cdfe5979fab00dea5fe6

`dist` folder now looks like
```
.
├── include
│   └── openassetio_mediacreation
│       ├── openassetio_mediacreation.hpp
│       └── traits
│           ├── content
│           │   └── LocatableContentTrait.hpp
│           ├── content.hpp
│           ├── managementPolicy
│           │   ├── ManagedTrait.hpp
│           │   └── ResolvesFutureEntitiesTrait.hpp
│           ├── managementPolicy.hpp
│           ├── timeline
│           │   ├── ClipTrait.hpp
│           │   ├── TimelineTrait.hpp
│           │   └── TrackTrait.hpp
│           ├── timeline.hpp
│           └── traits.hpp
└── lib
    └── cmake
        └── OpenAssetIO-MediaCreation
            ├── OpenAssetIO-MediaCreationConfig.cmake
            ├── OpenAssetIO-MediaCreationConfigVersion.cmake
            └── OpenAssetIO-MediaCreationTargets.cmake
  ```